internal/charmstore/addentity.go

```diff
@@ -527,6 +527,10 @@ func (s *Store) addCharm(c charm.Charm, p addParams) (err error) {
 		CharmRequiredInterfaces: interfacesForRelations(c.Meta().Requires),
 		SupportedSeries:         c.Meta().Series,
 	}
+	metrics := c.Metrics()
+	if metrics != nil && len(metrics.Metrics) > 0 {
```

Comment:
I think it's worth being defensive here, as done in other parts pf the system. AFAIK it's unlikely that another field wil be added to metrics in the near future.